stmhal/modmachine.c

```diff
@@ -207,49 +207,49 @@ STATIC mp_obj_t machine_freq(mp_uint_t n_args, const mp_obj_t *args) {
             // use HSI as SYSCLK
             sysclk_source = RCC_SYSCLKSOURCE_HSI;
         } else {
-            // search for a valid PLL configuration that keeps USB at 48MHz
-            for (; wanted_sysclk > 0; wanted_sysclk--) {
+            // sysclk = (HSE_VALUE * n) / (m * p). pll48ck =  (HSE_VALUE * n) / (m * q). We want pll48ck at 48Mhz
+            // valid sysclk and p found at or below wanted_sysclk
+            uint32_t target_sysclk = 0, target_p = 0;
+            for (q = 2; q <= 15; q++) {
+                // compute VCO_OUT with pll48ck at 48Mhz
+                uint32_t vco_out = 48 * q;
+                // make sure VCO_OUT is between 192MHz and 432MHz
+                if (vco_out < 192 || vco_out > 432) {
+                    continue;
```

Comment:
It seems that if you range q from 4 to 9 then vco_out will always be between 192 and 432 and you can remove this test.